### PR TITLE
stop changing labels, the template is responsible for getting these r…

### PR DESCRIPTION
**Pull request by grosser**
…ight

 - validate deploy/template/service labels are in sync (at deploy time)
 - validate labels make the service/deploy unique (at deploy time)

@zendesk/paas @rata

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb [hunk 1]
```diff
@@ -128,15 +96,20 @@ def set_docker_image
...
     end
 
+    # helpful env vars, also useful for log tagging
     def set_env
       env = (container.env || [])
 
       # static data
       metadata = release_doc_metadata
-      [:REVISION, :TAG, :PROJECT, :ROLE, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
+      [:REVISION, :TAG, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
```
Comment (rata) [inline plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb]: Not sure I follow, but does means that the rev, tag, etc. will be added as labels, and the yaml file should have the project and roles labels in the template for the pods that will be created?
Not that I have a problem with that, just want to be sure I understand (sorry, I'm a newby in web and ruby :))

Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb]: - rev/tag/etc will only be added to the pods and not the deployments/services/selectors
 - existing tags will not be overirtten
 - template is responsible for setting consistent role/project labels everywhere

sounds good ?

Comment (rata) [inline plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb]: On Wed, May 18, 2016 at 08:55:20AM -0700, Michael Grosser wrote:
> >      def set_env
> >        env = (container.env || [])
> >  
> >        # static data
> >        metadata = release_doc_metadata
> > -      [:REVISION, :TAG, :PROJECT, :ROLE, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
> > +      [:REVISION, :TAG, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
> 
>  - rev/tag/etc will only be added to the pods and not the deployments/services/selectors
>  - existing tags will not be overirtten
>  - template is responsible for setting consistent role/project labels everywhere
> 
> sounds good ?

Perfect! :-)

**Comment by jonmoter** [comment]
Sure, let's give it a try.  👍

**Pull request merged**